zerver/tests/test_events.py

```diff
@@ -2780,10 +2780,37 @@ def do_set_realm_property_test(self, name: str) -> None:
         )
 
         vals = test_values.get(name)
+        group_permission_setting_name = name[:-3]
```

Comment:
This is ugly code; I think you want something that explicitly removes `"_id"` if present at the end of the setting name; there's a number of reasonable Pythonic ways to do that. What you've written behaves unexpectedly if a setting name is 2 characters long, and is also not readable.